tenacity/retry.py

```diff
@@ -111,6 +112,61 @@ def __call__(self, attempt):
             return not self.predicate(attempt.result())
 
 
+class retry_if_exception_message(retry_base):
+    """Retries if an exception message equals or matches."""
+
+    def __init__(self, message=None, match=None, require_exception=False):
+        if message and match:
+            raise TypeError(
+                "{}() takes either 'message' or 'match', not both".format(
+                    self.__class__.__name__))
+        self.predicate = self._predicate_factory(message, match)
+        self.require_exception = require_exception
```

Comment:
That does not seem to be related to the heart of the function. I'd remove that, and put that in a different class (and PR) if that's needed.